test/parallel/test-beforeexit-event.js

```diff
@@ -1,42 +1,41 @@
 'use strict';
-require('../common');
-var assert = require('assert');
-var net = require('net');
-var revivals = 0;
-var deaths = 0;
+const common = require('../common');
+const net = require('net');
 
-process.on('beforeExit', function() { deaths++; });
+const beforeExit = common.mustCall(() => {
+  return;
+}, 4);
+
+const beforeListener = common.mustCall(() => {
+  return;
+}, 3);
+
+process.on('beforeExit', beforeExit);
 
 process.once('beforeExit', tryImmediate);
 
 function tryImmediate() {
-  console.log('set immediate');
-  setImmediate(function() {
-    revivals++;
+  setImmediate(() => {
+    beforeListener();
     process.once('beforeExit', tryTimer);
   });
 }
 
 function tryTimer() {
-  console.log('set a timeout');
-  setTimeout(function() {
-    console.log('timeout cb, do another once beforeExit');
-    revivals++;
+  setTimeout(() => {
+    beforeListener();
     process.once('beforeExit', tryListen);
   }, 1);
 }
 
 function tryListen() {
-  console.log('create a server');
   net.createServer()
     .listen(0)
     .on('listening', function() {
-      revivals++;
+      beforeListener();
       this.close();
+    })
+    .on('error', (e) => {
```

Comment:
This `'error'` handler can be removed since node will already throw if there are no `'error'` handlers.